### DISCUSSION TITLE
How do you run helloSystem?

### DISCUSSION THREAD

**Discussion by grd** [helloSystem/hello#307] [Q&A] [answered]
Q1: Are you guys running hello on bare metal or with a VM? What is the best approach? 

My daily driver is xubuntu (and ubuntu). I know, I know. It has its faults but it works. I have a Thinkpad T440p and 8Gb ram and a SSD 500Gb hd. Is that "supported"? In ubuntu there are no additional drivers, the Intel drivers just work. Should I use ZFS?

Q2: How stable is hello? Does it crash regularly (daily, weekly), and can you restore with ZFS?

Q3: Is hello only designed to run a desktop or also as a server? For instance running Nextcloud as a server.

Q4: I think I love this system, but what are the downsides of hello or FreeBSD?

Q5: How do you backup?

**Comment by grd** [comment, marked as answer]
I see it now after I watched your video on youtube. You use bare metal.

You can close my questions.

**Reply by probonopd** [reply]
Welcome @grd

> Q1: Are you guys running hello on bare metal or with a VM? What is the best approach?

I've been running helloSystem as my daily driver on bare metal since over a year. Before that I has been using mainly Xubuntu and occasionally other Linux distributions.

> I have a Thinkpad T440p and 8Gb ram and a SSD 500Gb hd. Is that "supported"?

I think this is quite adequate hardware. The SD card reader, smartcard reader and fingerprint sensor may not work. If you can do without these, it should work rather well. There is a very easy way to test it - write the Live ISO to a USB stick and boot from that.

> In ubuntu there are no additional drivers, the Intel drivers just work.

> Should I use ZFS?

Yes. (You are not even asked about this when you install helloSystem.)

> Q2: How stable is hello? Does it crash regularly (daily, weekly), and can you restore with ZFS?

It never crashes on me when I am doing normal work. When testing out unusal (unsupported) things during development I did get occasional kernel panics, but it never crashed on me when just using the computer like a regular user.

> Q3: Is hello only designed to run a desktop or also as a server? For instance running Nextcloud as a server.

helloSystem is designed as a desktop operating system welcoming to non-technical users. But it is still FreeBSD under the hood, which is an extremely powerful server operating system. So you should not run into issues running a Nextcloud server on it, as long as you are fine with installing and configuring it via the command line (something "mere mortals" don't even know exists).

> Q4: I think I love this system, but what are the downsides of hello or FreeBSD?

* FreeBSD: Limited hardware compatibility compared to Windows and Linux, but still more hardware compatibility compared to macOS. This is mainly an issue when using very new or exotic hardware. One area that is frequently cited as problematic is WLAN (especially Intel) driver support, but this is being worked on in the FreeBSD project
* helloSystem: System is still new, many components have not yet left the "Under Construction" status (but to change this, we need more users who are also willing to test and send patches). Since helloSystem has FreeBSD under the hood, most of the time you can use the command line in case a GUI utility does not yet do what you need it to

> Q5: How do you backup?

Personally I copy my home directory to an external drive from time to time. FreeBSD offers way more advanced backup solutions (e.g., involving zfs) but I have not taken the time yet to wrap my head fully around them.

**Reply by grd** [reply]
> I think this is quite adequate hardware. The SD card reader, smartcard reader and fingerprint sensor may not work. If you can do without these, it should work rather well. There is a very easy way to test it - write the Live ISO to a USB stick and boot from that.

The SD card reader is essential to me since I use it to read and write my rpi. For the rest, I don't care about the fingerprint stuff nor smartcard reader. I saw that FreeBSD supports a SD card, so it could and should work.

> Personally I copy my home directory to an external drive from time to time. FreeBSD offers way more advanced backup solutions (e.g., involving zfs) but I have not taken the time yet to wrap my head fully around them.

I can understand that. I usually just tar the system. It is a matter of taste.

Oke. I have one more question. Does everything with an AppImage "just work"?

**Reply by probonopd** [reply]
> Does everything with an AppImage "just work"?

Not quite yet. But we are working on it.

**Reply by grd** [reply]
You convinced me. I am gonna try and report back later. (after a small well deserved holiday ;-) )

**Reply by alphamodh0** [reply]
As my only installed OS.
